### ISSUE TITLE
Support consuming wado-rs endpoint with cross-origin resource sharing (CORS)  

### ISSUE THREAD

**Issue by Zaid-Safadi** [microsoft/dicom-server#2021]
**User story**
As a server consumer, I want the DICOMweb server to support consuming its API with CORS so that web viewers hosted outside the server domain can consume the server endpoint.

Currently, when calling the wado-rs endpoint to retrieve a frame from an SPA client (OHIF Viewer). The browser fails with a CORS error:
`Access to XMLHttpRequest at 'https://xxxxxxxx.azurewebsites.net/v1/studies/1.3.../series/1.3.../instances/1.3..../frames/1' from origin 'http://localhost:3000' has been blocked by CORS policy: Request header field accept is not allowed by Access-Control-Allow-Headers in preflight response.`

Note this is "Access-Control-Allow-Headers" which I don't believe can be configured from the Azure portal App Service CORS blade, hence may require support this from the server.

Right now, it is not possible to run a viewer from "localhost" due to this:
![image](https://user-images.githubusercontent.com/10813109/197076718-eefbfbe4-6985-4842-9388-ab4e020bad22.png)

![image](https://user-images.githubusercontent.com/10813109/197077619-e8db0c31-e99e-4713-8edc-d30a6733afea.png)

**Acceptance criteria**
1. When I run an SPA viewer from a different domain than the server, then the server would support consuming the WADO-RS frame endpoint.
When running the same request with chrome flag: disable-web-security it works as expected and open the image

**Comment by Zaid-Safadi** [comment]
Here is an example of how this can be enabled in the server:

https://github.com/DICOMcloud/DICOMcloud/blob/development/DICOMcloud.Wado.WebApi/Types/CorsPolicyFactory.cs
https://github.com/DICOMcloud/DICOMcloud/blob/e7b32a98567d1e718b615ab5004edde49079964d/DICOMcloud.Wado.WebApi/App_Start/WebApiConfig.cs#L23

**Comment by smithago** [comment]
Probably we need to document how to enable CORS. Here is how you can do it.
Edit this file https://github.com/microsoft/dicom-server/blob/main/src/Microsoft.Health.Dicom.Web/appsettings.json and add the CORS settings there like below and run the service.
![image](https://user-images.githubusercontent.com/6980724/197084377-f6aff1bd-c22f-4610-8ca4-e6e6215ba2b5.png)

"Cors": {
   "Origins": ["*"],
   "Methods": ["*"],
   "Headers": ["*"],
   "MaxAge": 1440
}

**Comment by Zaid-Safadi** [comment]
@smithago, I've been using the ARM template to deploy to azure so I didn't have a local dev environment. 

First, I tried updating this from the Azure portal config but didn't seem to make a difference.
DicomServer__Cors__Headers      *
DicomServer__Cors__Methods     *
DicomServer__Cors__Origins        *

I then cloned the repo and looked at the project source code, but can't find how the cors configuration "CorsConfiguration" is being used in the project. The only reference I see for cors is enabling a default policy in "DicomServerServiceCollectionExtensions":
app.UseCors(CorsConstants.DefaultCorsPolicy);

I am probably missing something but I wanted before creating a local dev environment and deploy that from my machine to quickly check with you on these observations and how cors is enabled. 

Here is my config in the Azure portal:
![image](https://user-images.githubusercontent.com/10813109/197677657-b4a56016-1ce1-4dd7-a9b4-516ad0a7d50e.png)
